### PR TITLE
Handle sunrise/sunset times that occur on the previous/next day

### PR DESCRIPTION
**Pull request by c-mckenna**
Hey @harry093,

Bill asked me to sync this repo up with the latest changes made to the internal repo.

Can you please take a look, this was the fix to address GEOD-103

Cheers,
Callum

### CODE REVIEW

**Review by harry093 — APPROVED** [review]
@c-mckenna good catch, I didn't occur to me that that fix would affect the base code. I am happy to merge this and create another version of the code. Am I good to do that?

**Comment by c-mckenna** [comment]
Thanks @harry093,

Yep this is all good to merge now, I don't have write access so I'll leave that to you.

**Pull request merged**
